packages/server/src/integration-test/mysql.spec.ts

```diff
@@ -0,0 +1,1289 @@
+import fetch from "node-fetch"
+import {
+  generateMakeRequest,
+  MakeRequestResponse,
+} from "../api/routes/public/tests/utils"
+
+import * as setup from "../api/routes/tests/utilities"
+import {
+  Datasource,
+  FieldType,
+  RelationshipType,
+  Row,
+  SaveTableResponse,
+  Table,
+  TableRequest,
+  TableSourceType,
+} from "@budibase/types"
+import _ from "lodash"
+import { generator } from "@budibase/backend-core/tests"
+import { utils } from "@budibase/backend-core"
+import { databaseTestProviders } from "../integrations/tests/utils"
+import mysql from "mysql2/promise"
+import { builderSocket } from "../websockets"
+// @ts-ignore
+fetch.mockSearch()
+
+const config = setup.getConfig()!
+
+jest.unmock("mysql2/promise")
+jest.mock("../websockets", () => ({
+  clientAppSocket: jest.fn(),
+  gridAppSocket: jest.fn(),
+  initialise: jest.fn(),
+  builderSocket: {
+    emitTableUpdate: jest.fn(),
+    emitTableDeletion: jest.fn(),
+    emitDatasourceUpdate: jest.fn(),
+    emitDatasourceDeletion: jest.fn(),
+    emitScreenUpdate: jest.fn(),
+    emitAppMetadataUpdate: jest.fn(),
+    emitAppPublish: jest.fn(),
+  },
+}))
+
+describe("mysql integrations", () => {
+  let makeRequest: MakeRequestResponse,
+    mysqlDatasource: Datasource,
+    primaryMySqlTable: Table,
+    oneToManyRelationshipInfo: ForeignTableInfo,
+    manyToOneRelationshipInfo: ForeignTableInfo,
+    manyToManyRelationshipInfo: ForeignTableInfo
+
+  beforeAll(async () => {
+    await config.init()
+    const apiKey = await config.generateApiKey()
+
+    makeRequest = generateMakeRequest(apiKey, true)
+
+    mysqlDatasource = await config.api.datasource.create(
+      await databaseTestProviders.mysql.datasource()
+    )
+  })
+
+  afterAll(async () => {
+    await databaseTestProviders.mysql.stop()
+  })
+
+  beforeEach(async () => {
+    async function createAuxTable(prefix: string) {
+      return await config.createTable({
+        name: `${prefix}_${generator.word({ length: 5 })}`,
```

Comment:
This may not be random enough, and could cause flakes. Any reason to not use a UUID here?

Reply:
I actually had to reduce this from 6 -> 5 because MySQL was complaining about the identifier lengths. MySQL is limited to 64 characters, so longer join table identifiers were problematic.

Reply:
Looking at the usages of `createAuxTable`, I would personally get rid of the `prefix` argument and make the name always a UUID. A UUID is 36 chars long and shouldn't ever clash.

Been wondering about this for a while so decided to write a little script to see.

```javascript
const uuid = require("uuid")
const chance = require("chance")
const c = new chance.Chance()

const words = new Set()
const uuids = new Set()

const N = 10000000
for (let i = 0; i < N; i++) {
  words.add(c.word({ length: 5 }))
  uuids.add(uuid.v4())
}

function uniqueValues(set) {
  return `${set.size}/${N} (${((set.size / N) * 100).toFixed(2)}%)`
}

console.log(`words uniqueness: ${uniqueValues(words)}`)
console.log(`uuids uniqueness: ${uniqueValues(uuids)}`)
```

Output:

```
words uniqueness: 834493/10000000 (8.34%)
uuids uniqueness: 10000000/10000000 (100.00%)
```

Sort of a flawed test, it looks like the number of possible unique values of the 5 letter words `chance` generates tops out around ~800,000, so comparing against such a large N is silly. But the point is the same: UUIDs will collide way less often.